frontend/components/Course/InstructorQueuePage/Queue.tsx

```diff
@@ -57,6 +64,37 @@ const Queue = (props: QueueProps) => {
 
     const [clearModalOpen, setClearModalOpen] = useState(false);
 
+    const handlePinChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e.target.value.length <= PIN_CHAR_LIMIT && editingPin && active) {
+            setPinState(e.target.value);
+        }
+    };
+
+    const handlePinKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (editingPin && e.key === "Enter") {
+            setPinMutated(false);
+            setEditingPin(false);
+            partialUpdateQueue(courseId, queueId, { pin: pinState });
+            mutate(pin, { pin: pinState }).then(() => setPinMutated(true));
+        } else if (editingPin && e.key === "Escape") {
+            setPinState(pin);
+            setEditingPin(false);
+        }
+    };
+
+    const handlePinToggle = () => {
+        if (editingPin) {
+            setPinMutated(false);
+            setEditingPin(false);
+            partialUpdateQueue(courseId, queueId, { pin: pinState });
+            mutate(pin, { pin: pinState }).then(() => setPinMutated(true));
```

Comment:
Important - From my testing, the changed pin does not display correctly after you edit it

What are we using setPinMutated for?